src/main/java/rx/internal/util/UtilityFunctions.java

```diff
@@ -61,6 +64,56 @@ public T call(T o) {
         };
     }
 
+    /**
+     * Waits for completion of the Observable/Single and stops the subscription in case of error.
+     *
+     * @param latch
+     *         latch that waits for completion of the Observable/Single.
+     * @param subscription
+     *         subscription that needs to be unsubscribed in case of error.
+     */
+    public static void awaitForCompletion(CountDownLatch latch, Subscription subscription) {
```

Comment:
I'd put these into a separate class, `BlockingUtils` or `BlockingHelper` instead of the `UtilityFunctions` for better (internal) discoverability.